src/Request.js

```diff
@@ -245,17 +246,23 @@ export default class Request {
 
     retry () {
 
-        if (this.retries++ < this._max_retry) {
+        if (this.last_error) {
+            this.errors.push(this.last_error);
+        }
+
+        if (this.retries + 1 < this._max_retry) {
```

Comment:
I think that'll actually be a rare case, but I think it should be achieved by having a function that 

.add_retries(3)

And on which, it'll do this._max_retry = this.retries + new_retries;

So we keep an accurate count of how many times we actually retried.

And we can also have reset_retries etc.,